### ISSUE TITLE
Support zipimport

### ISSUE THREAD

**Issue by Pluckerpluck** [ZoomerAnalytics/xlwings#1006]
So I'm using xlwings, and would like to make my package a little more convenient to distribute.
As a result, I would love to be able to zip up my python code before distribution, just to keep it nicely contained.

Python can natively import modules from a zip archive, and you can already set `PYTHONPATH` to include the zip archive, but currently this causes the code to crash, as it tries to find the time files within the archive were last modified.

There are two options:

1. Assume that anything with `.zip` in the path just won't change. This is an easy fix (I think) and the following function in `udfs.py` works to achieve this:

```python
def get_udf_module(module_name):
    module_info = udf_modules.get(module_name, None)
    if module_info is not None:
        module = module_info['module']
        if module_info['filetime'] == None: # If is .zip file
            return module # Assume it hasn't changed
        mtime = os.path.getmtime(module_info['filename'])
        if mtime == module_info['filetime']:
            return module
        else:
            module = reload(module)
            module_info['filetime'] = mtime
            module_info['module'] = module
            return module
    else:
        if sys.version_info[:2] < (2, 7):
            # For Python 2.6. we don't handle modules in subpackages
            module = __import__(module_name)
        else:
            module = import_module(module_name)
        filename = os.path.normcase(module.__file__.lower())
        if ".zip" not in filename: 
            mtime = os.path.getmtime(filename)
        else:
            mtime = None
        
        udf_modules[module_name] = {
            'filename': filename,
            'filetime': mtime,
            'module': module
        }
        return module
```

2. Detect if it's a zip archive (again, just look for `.zip`), and then check the modification time on the zip file, and not the module itself.

    Unfortunately, this does require reloading all the modules in the zip file though, which is a massive pain, especially as `reload()` doesn't work. So I suggest option 1.

Only reason I haven't pull requested this is I haven't had time to ensure there are no other side effects, as I use a modification time of `None` to indicate a zip file.

**Comment by fzumstein** [comment]
Option 1 sounds reasonable to me, feel free to open a PR and I can test it out

**Comment by fzumstein** [comment]
released with 0.15.2
